packages/clay-date-picker/src/index.tsx

```diff
@@ -631,14 +649,12 @@ const ClayDatePicker = React.forwardRef<HTMLInputElement, IProps>(
 							alignElementRef={triggerElementRef}
 							className="date-picker-dropdown-menu"
 							data-testid="dropdown"
+							id={ariaControls}
 							onActiveChange={setExpandedValue}
 							ref={dropdownContainerRef}
+							role="dialog"
```

Comment:
Might be worthwhile to add an 'aria-label' to the dialog element, since then the screen reader will announce it when the dialog is focused. For example: "Choose Date Modal Dialog" will be read when the aria-label="Choose Date".